### PR TITLE
Update build to use VS2015

### PR DESCRIPTION
**Pull request by sidhoda**
Includes updating appveyor.yml.

ASIO is now a submodule so updated to init submodules during install, replacing the download step.
OpenSSL is already installed on the build worker, so download step now redundant.

Will need clean-up to remove vs2015 branch from appveyor.yml.

### CODE REVIEW

**Comment by sidhoda** [comment]
Should be all set to merge this now?

**Comment by jadamcrain** [comment]
I'll look into this early next week. I've been busy on the development/iorefactor branch. Unfortunately, you'll probably want to repeat your testing, but the goods news is that we have TLS integration tests and I've been able to simply things greatly by merging the GPRS/traditional APIs onto a single "DNP3Manager".

File: build-dotnet.bat [hunk 1]
```diff
@@ -20,19 +20,19 @@ IF ["%OSSL_LIB32_DIR%"]==[] (
...
 :BUILD
-CALL "%VS120COMNTOOLS%VsDevCmd.bat"
+CALL "%VS140COMNTOOLS%VsDevCmd.bat"
 
 IF EXIST build RMDIR build /s /q
 MKDIR build\lib
 CD build
 
-cmake .. -DCMAKE_INSTALL_PREFIX=lib -DFULL=ON -DCMAKE_BUILD_TYPE=%CONFIGURATION% -G"Visual Studio 12 2013"
+cmake .. -DCMAKE_INSTALL_PREFIX=lib -DFULL=ON -DCMAKE_BUILD_TYPE=%CONFIGURATION% -G"Visual Studio 14 2015"
```
Comment (sidhoda) [inline build-dotnet.bat]: The FULL option has been changed on the development branch to DNP3_ALL.

As the development branch now needs VS2015, would it be worth rebasing this?

**Comment by jadamcrain** [comment]
Yes, rebasing the tweaks to the build script and the appveyor config onto development would make things easiest for me.

**Comment by sidhoda** [comment]
Not sure if I've done this right by merging rather than rebasing. Let me know if it will make a mess of things and I'll redo.

**Comment by sidhoda** [comment]
OK, I've rebased onto development which has cleaned up the git history.

**Pull request merged**